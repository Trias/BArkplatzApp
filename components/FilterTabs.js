import React from 'react';
import {ScrollView, TouchableOpacity, Text, StyleSheet} from 'react-native';

const filterMappping = {
    'Normal': 'Standard',
    'Standard': 'Standard',
    'Elektro': 'Elektro',
    'Mutter-Kind': 'Mutter-Kind',
    'Versehrtenparkplatz': 'Behindertenparkplatz',
    'Behindertenparkplatz': 'Behindertenparkplatz',
    'Alle': 'Alle'
}
const filterToInternalMap = {
    'Standard': 'Normal',
    'Elektro': 'Elektro',
    'Mutter-Kind': 'Mutter-Kind',
    'Behindertenparkplatz': 'Versehrtenparkplatz',
    'Alle': 'Alle'
};

export function FilterTabs({ filters, onFilterPress, activeFilter }) {
    return <ScrollView style={styles.tabs} horizontal={true} showsHorizontalScrollIndicator={false}>
        {filters.map((filter) => {
            return <TouchableOpacity 
            key={filter} 
            onPress={() => onFilterPress(filterToInternalMap[filter])} 
            style={[styles.tab, filterMappping[activeFilter] == filter ? styles.activeTab : null]
            }>
                <Text style={[styles.filter, filterMappping[activeFilter] == filter ? styles.activeFilter : null]}>{filter}</Text>
            </TouchableOpacity>
        })}
    </ScrollView>
}

const styles = StyleSheet.create({
    tabs: {
        height: 60,
        lineHeight: 60,
        flex: 1,
        position: 'absolute',
        zIndex: 100,
        backgroundColor: 'white',
        left: 0,
        right: 0,
    },
    tab: {
        paddingHorizontal: 5,
        paddingVertical: 10,
        marginHorizontal: 5,
        borderBottomWidth: 3,
        borderBottomColor: 'transparent',
    },
    filter: {
        fontWeight: 'bold',
        fontSize: 14,
        marginVertical: 10,
    },
    activeTab: {
        borderBottomColor: 'red',
    },
    activeFilter: {
        color: 'red',
    },
})