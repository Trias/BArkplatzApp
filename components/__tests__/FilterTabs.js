import React from 'react';
import renderer from 'react-test-renderer';

import {FilterTabs} from '../FilterTabs';

it(`renders correctly`, () => {
  const tree = renderer.create(<FilterTabs 
    filters={['Alle', 'Elektro']}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
