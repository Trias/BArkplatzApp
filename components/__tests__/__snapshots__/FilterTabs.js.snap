// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  horizontal={true}
  showsHorizontalScrollIndicator={false}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "height": 60,
      "left": 0,
      "lineHeight": 60,
      "position": "absolute",
      "right": 0,
      "zIndex": 100,
    }
  }
>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderBottomColor": "transparent",
          "borderBottomWidth": 3,
          "marginHorizontal": 5,
          "opacity": 1,
          "paddingHorizontal": 5,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 14,
              "fontWeight": "bold",
              "marginVertical": 10,
            },
            null,
          ]
        }
      >
        Alle
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderBottomColor": "transparent",
          "borderBottomWidth": 3,
          "marginHorizontal": 5,
          "opacity": 1,
          "paddingHorizontal": 5,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 14,
              "fontWeight": "bold",
              "marginVertical": 10,
            },
            null,
          ]
        }
      >
        Elektro
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
